pkg/provider/aws/resources/securitygroup.go

```diff
@@ -0,0 +1,46 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/config"
+	"github.com/klothoplatform/klotho/pkg/core"
+)
+
+type SecurityGroup struct {
+	Name          string
+	Vpc           *Vpc
+	ConstructsRef []core.AnnotationKey
+}
```

Comment:
Should have Ingress and Egress rules here so we can add to them and modify dynamically

Reply:
Would you be fine with leaving that for a future change when we want to make them dynamic?

Reply:
Sure although i think we do it as a fast follow because as we do the eks work it being dynamic will be beneficial

Reply:
Can you create a task for this? Becuase the eks cluster will definitely need to add its own rule to the SG